### PR TITLE
Fix the implementation of the nn sum extended FEEL function for edge cases

### PR DESCRIPTION
**Pull request by SimonRinguette**
Fixed cases are:

- nn sum( [] ) => returned null, now returns 0
- nn sum( [null] ) => exception thrown, now returns 0
- nn sum( [null, null] ) => returned null, now returns 0

The BaseFEELFunction.java fix could also fix other cases where a list containing a single null element was used in a list.

### CODE REVIEW

**Comment by SimonRinguette** [comment]
@tarilabs as discussed in an email discussion, here is the PR that fix the behaviour of the nn sum feel method to more closelly follow the sum feel method.

**Comment by tarilabs** [comment]
Jenkins test again

**Comment by tarilabs** [comment]
ok to test

**Comment by tarilabs** [comment]
Jenkins execute compile downstream build

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java [hunk 1]
```diff
@@ -35,6 +35,9 @@
                 { "now()", ZonedDateTime.class , null},
                 { "today()", LocalDate.class, null },
                 { "nn sum( 10, null, 20, 40, null )", new BigDecimal("70", MathContext.DECIMAL128), null },
+                { "nn sum( [] )", new BigDecimal("0", MathContext.DECIMAL128), null },
+                { "nn sum( [ null ] )", new BigDecimal("0", MathContext.DECIMAL128), null },
+                { "nn sum( [ null, null ] )", new BigDecimal("0", MathContext.DECIMAL128), null },
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java]: ok

**Review by tarilabs — APPROVED** [review]
approving when green

**Review by tarilabs — CHANGES_REQUESTED** [review]
sorry just to be clear @SimonRinguette 
this will make that
`nn sum([]) = 0`
while from the spec we have
`sum([]) = null`
just to confirm the different behaviour is what is expected, we shall review this in the thread with Bruce?

**Comment by SimonRinguette** [comment]
@tarilabs This was because I was using a build that did not contain your https://issues.redhat.com/browse/DROOLS-5440 fix, updating my PR in a few minutes

**Comment by SimonRinguette** [comment]
@tarilabs I forced pushed the commit to change the message.

While this fix nn sum([null]), i'm pretty sure that it also fixes other edge cases with method with multiple signatures that can receive a null list in the parameter and would fail similarly

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java [hunk 2]
```diff
@@ -35,6 +35,9 @@
                 { "now()", ZonedDateTime.class , null},
                 { "today()", LocalDate.class, null },
                 { "nn sum( 10, null, 20, 40, null )", new BigDecimal("70", MathContext.DECIMAL128), null },
+                { "nn sum( [] )", null, null },
+                { "nn sum( [ null ] )", null, null },
+                { "nn sum( [ null, null ] )", null, null },
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java]: Consistent with `sum([]) = null` from DMN spec (v1.3) Table 75, good for me thanks!

**Review by tarilabs — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java (hunk 2 above)
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java]:
```suggestion
                { "nn sum( [] )", BigDecimal.ZERO, null },
                { "nn sum( [ null ] )", BigDecimal.ZERO, null },
                { "nn sum( [ null, null ] )", BigDecimal.ZERO, null },
```

**Review by tarilabs — CHANGES_REQUESTED** [review]
based on conversation

**Comment by SimonRinguette** [comment]
@tarilabs Hopefully wrapping this one up... we have done some back and forth.

The PR is now back to the original Use Cases described in the issue as discussed between me / you / @etirelli and Bruce Silver in a separate email thread.

We decided to have nn sum([]) = 0 instead of nn sum([]) = null. nn sum([]) = null would be more aligned to the existing sum([]) = null FEEL function but the business requirement for this extension function is to ignore the notion of nulls and make the sum 0.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java [hunk 3]
```diff
@@ -35,6 +35,9 @@
                 { "now()", ZonedDateTime.class , null},
                 { "today()", LocalDate.class, null },
                 { "nn sum( 10, null, 20, 40, null )", new BigDecimal("70", MathContext.DECIMAL128), null },
+                { "nn sum( [] )", BigDecimal.ZERO, null },
+                { "nn sum( [ null ] )", BigDecimal.ZERO, null },
+                { "nn sum( [ null, null ] )", BigDecimal.ZERO, null },
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java:40]: 👍  as discussed

**Review by tarilabs — APPROVED** [review]

**Comment by tarilabs** [comment]
will ask for merge since this implements required changes https://github.com/kiegroup/drools/pull/3012#pullrequestreview-464019201 as discussed
